web/src/engine/main/src/contextManagerBase.ts

```diff
@@ -1,11 +1,44 @@
 import EventEmitter from 'eventemitter3';
-import { type Keyboard, type KeyboardInterface, type OutputTarget } from '@keymanapp/keyboard-processor';
-import { type KeyboardStub } from 'keyman/engine/package-cache';
+import { DeviceSpec, ManagedPromise, type Keyboard, type KeyboardInterface, type OutputTarget } from '@keymanapp/keyboard-processor';
+import { StubAndKeyboardCache, type KeyboardStub } from 'keyman/engine/package-cache';
 import { PredictionContext } from '@keymanapp/input-processor';
 
 interface EventMap {
   // target, then keyboard.
-  'targetchange': (target: OutputTarget) => void;
+  'targetchange': (target: OutputTarget) => boolean;
+
+  /**
+   * This event is raised whenever a keyboard change is requested.  Calling the second parameter -
+   * `abortChange` - will abort the process and prevent the change.
+   *
+   * Note that if the keyboard has not been previously loaded, this event will be raised twice.
+   * 1. Before the keyboard is loaded into Keyman Engine for Web.
+   *     - Aborting at this point will prevent the keyboard from being loaded, with no network
+   *       request for the keyboard resource being triggered.
+   * 2. Once the keyboard is loaded, but before it is activated.
```

Comment:
Cross-reference with https://help.keyman.com/DEVELOPER/ENGINE/WEB/15.0/reference/events/kmw.beforekeyboardchange.

`abortChange` is the corresponding fix for #8482.

The "event raised twice" bit is based on the existing implementation of the event within KMW 16.0 and the timings thereof:

Check 1:
https://github.com/keymanapp/keyman/blob/f33f31521ae743ba6f9f0775965c86d1d28bfb79/web/src/engine/main/keyboards/kmwkeyboards.ts#L465
https://github.com/keymanapp/keyman/blob/f33f31521ae743ba6f9f0775965c86d1d28bfb79/web/src/engine/main/keyboards/kmwkeyboards.ts#L475-L476

(Where `_SetActiveKeyboard` is what triggers the keyboard-activation process)

Check 2:

https://github.com/keymanapp/keyman/blob/f33f31521ae743ba6f9f0775965c86d1d28bfb79/web/src/engine/main/keyboards/kmwkeyboards.ts#L713
https://github.com/keymanapp/keyman/blob/f33f31521ae743ba6f9f0775965c86d1d28bfb79/web/src/engine/main/keyboards/kmwkeyboards.ts#L752-L753
https://github.com/keymanapp/keyman/blob/f33f31521ae743ba6f9f0775965c86d1d28bfb79/web/src/engine/main/keyboards/kmwkeyboards.ts#L766-L768

(Upon the asynchronous loading of a newly-activating keyboard; `installKeyboard` is called by `_SetActiveKeyboard` for this case, listed as case 4 in the PR description)

Reply:
... now that I've looked deeper into how the older style of KMW events operates... I think I was mistaken here.  That same verbiage is used for other events that clearly don't cancel any ongoing process whatsoever.  So... accidental new feature, if we leave it in.

I spent so much time to make that work, too...